src/main/java/com/questhelper/helpers/quests/thecurseofarrav/TheCurseOfArrav.java

```diff
@@ -0,0 +1,786 @@
+		inZemouregalsBaseKitchen = new ZoneRequirement(new Zone(new WorldPoint(3613, 4604, 0), new WorldPoint(3606, 4598, 0)));
+		inZemouregalsBaseSewer = new ZoneRequirement(new Zone(14919));
+	}
+
+	@Override
+	protected void setupRequirements()
+	{
+		haveKilledGolem = new QuestRequirement(QuestHelperQuest.THE_CURSE_OF_ARRAV, 12);
+		haveMadeCanopicJar = new QuestRequirement(QuestHelperQuest.THE_CURSE_OF_ARRAV, 18);
+		haveMinedAFullPath = new QuestRequirement(QuestHelperQuest.THE_CURSE_OF_ARRAV, 30);
+		haveUsedPlans = new QuestRequirement(QuestHelperQuest.THE_CURSE_OF_ARRAV, 38);
+		haveUsedKey = new QuestRequirement(QuestHelperQuest.THE_CURSE_OF_ARRAV, 46);
+		var haveMetArrav = new QuestRequirement(QuestHelperQuest.THE_CURSE_OF_ARRAV, 54);
+
+		dwellberries3 = new ItemRequirement("Dwellberries", ItemID.DWELLBERRIES, 3);
+		dwellberries3.setConditionToHide(haveMadeCanopicJar);
+		ringOfLife = new ItemRequirement("Ring of life", ItemID.RING_OF_LIFE);
+		ringOfLife.setConditionToHide(haveMadeCanopicJar);
+		anyPickaxe = new ItemRequirement("Any pickaxe", ItemCollections.PICKAXES).isNotConsumed();
+		anyPickaxe.setConditionToHide(haveMinedAFullPath);
+		anyGrappleableCrossbow = new ItemRequirement("Any crossbow", ItemCollections.CROSSBOWS).isNotConsumed();
+		anyGrappleableCrossbow.setConditionToHide(haveMetArrav);
+		mithrilGrapple = new ItemRequirement("Mith grapple", ItemID.MITH_GRAPPLE_9419).isNotConsumed();
+		mithrilGrapple.setConditionToHide(haveMetArrav);
+		insulatedBoots = new ItemRequirement("Insulated boots", ItemID.INSULATED_BOOTS).isNotConsumed();
+
+		staminaPotion = new ItemRequirement("Stamina potion", ItemCollections.STAMINA_POTIONS, 1);
+		prayerPotion = new ItemRequirement("Prayer potion", ItemCollections.PRAYER_POTIONS, 1);
+		crushCombatGear = new ItemRequirement("Melee combat gear (crush preferred)", -1, -1);
+		crushCombatGear.setDisplayItemId(BankSlotIcons.getMeleeCombatGear());
+		crushCombatGear.setConditionToHide(haveKilledGolem);
+		rangedCombatGear = new ItemRequirement("Ranged combat gear for killing Arrav", -1, -1);
+		rangedCombatGear.setDisplayItemId(BankSlotIcons.getRangedCombatGear());
+		food = new ItemRequirement("Food", ItemCollections.GOOD_EATING_FOOD, -1);
+	}
+
+	public void setupSteps()
+	{
+		var todo = new NpcStep(this, 5, "TODO XD");
+
+		startQuest = new NpcStep(this, NpcID.ELIAS_WHITE, new WorldPoint(3505, 3037, 0), "Talk to Elias south of Ruins of Uzer (DLQ FAIRY RING).");
+		startQuest.addDialogStep("Yes.");
+
+		enterTomb = new ObjectStep(this, ObjectID.ENTRY_50201, new WorldPoint(3486, 3023, 0), "Enter the tomb south-west of Elias.");
+		// TODO: Ensure player can get hint to return
+
+		var insideTomb = new Zone(new WorldPoint(3842, 4603, 0), new WorldPoint(3900, 4547, 0));
+		var insideTombReq = new ZoneRequirement(insideTomb);
+		var hasFirstKey = new ItemRequirement("Mastaba Key", ItemID.MASTABA_KEY);
```

Comment:
I decided not to add this because:
1. It messes with the conditional step logic if the key is used as a condition
2. I think either key works for either lever (but I could be wrong with this)